src/useDebounce.ts

```diff
@@ -0,0 +1,37 @@
+import debounce from 'lodash.debounce';
+import {
+  useEffect,
+  useMemo,
+  useRef,
+  useCallback,
+} from 'react';
+
+export const useDebounce = <T extends (...args: any[]) => void>(
+  callback: T,
+  delay = 1000,
+): T => {
+  const callbackRef = useRef(callback);
+
+  useEffect(() => {
+    callbackRef.current = callback;
+  }, [callback]);
+
+  const debouncedCallback = useMemo(() => {
+    return debounce((...args: Parameters<T>) => {
+      callbackRef.current?.(...args);
+    }, delay);
+  }, [delay]);
+
+  useEffect(() => {
+    return () => {
+      debouncedCallback.cancel();
+    };
+  }, [debouncedCallback]);
```

Comment:
Looks too complicated, you can use `use-debounce` library with ready to use hook:
https://www.npmjs.com/package/use-debounce